[Feat] create retospect

yummygyudon opened the pull request:
<!-- pr 이름은 '[컨벤션] 기능이름' 으로 이슈와 통일해주세요. 이슈와 마찬가지로 라벨로 담장자를  표시해 주세요. ex. [Feat] searchPublicCourse -->

### 😶 무슨 이슈인가요?

---
#7 
<!-- 해당 pr과 연결된 이슈를 닫아주세요. closes #이슈넘버 -->

### 🤔 어떻게 이슈를 해결했나요?
- 잘못된 속성명으로 인식되어 JSON 데이터 매핑 → `@JsonProperty()` 로 명시하여 Mapping 해결
```java
@JsonProperty("isPublic")
```


---

- <!-- 실제로 해결한 로직을 써주세요, 이슈에 쓴 로직과 달라도, 또는 같아도 좋습니다. -->
-

### 🤯 주의할 점이 있나요?

---

<!-- 수정/추가한 내용중 주의깊게 봐야하는 부분이 있다면, 스크린샷으로 해당 부분을 자세히 설명해주세요. 그리고 왜 그렇게 변경했는지도 써주세요  -->

yummygyudon committed:
STYLE : 저장 메소드 파라미터 명 수정

- 기존 : `Retrospect user`
- 수정 : `Retrospect retrospect`

yummygyudon committed:
FEAT : 공통 사용 Class 및 Enum 코드 통일화

- ($common/dto) ApiResponse
- ($exception) Error
- ($exception) Success
- ($controller/dto/response) RetrospectResponseDto

yummygyudon committed:
FEAT : POST 성공 Status 정의

- Status : 201 Created
- Message : 회고 생성 성공

yummygyudon committed:
FEAT : POST 전용 Request Data 매핑 DTO 정의

- id 없는 상태의 Request에 맞춘 DTO
- JSON 속성명 Mapping 오류 @JsonProperty 어노테이션으로 해결

yummygyudon committed:
FIX : ResponseDTO 반환 JSON 속성 매핑 오류 해결

- 반환되는 객체 데이터 내 속성명이 `isPublic` 이 아닌 `public`으로 반환됨
- @JsonProperty 어노테이션을 통해 jackson 라이브러리에 속성명 명시

yummygyudon committed:
FIX : 신규 Retrospect 추가 메서드 정의 및 반환 타입 정정

- 정상적으로 추가되었는지 확인하기 위해 반환 타입 Retrospect 타입으로 변경

yummygyudon committed:
FEAT : 신규 Retrospect 추가 Service 메서드 정의

- Request Body 데이터 Converting
- 저장 결과를 Response Body 에 담길 RetrospectResponseDto로 Converting

yummygyudon committed:
FEAT : 신규 Retrospect 추가 POST Reqeust 매핑 Controller 정의

- Status : 201 CREATED
- 저장 결과 반환

funnysunny08 commented on server/src/main/java/sopt/mysoptin/server/controller/dto/request/RetrospectRequestCreateDto.java:
```diff
@@ -0,0 +1,21 @@
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class RetrospectRequestCreateDto {
+    @JsonProperty("isPublic")
+    private boolean isPublic;
```
아하 여기서 이렇게 한다는 거군요!! 👍

funnysunny08 reviewed (commented):
확인완료!!
수고하셨습니다:)👍

yummygyudon committed:
FEAT : PUT 성공 Status 정의

- Status : 200 OK
- Message : 회고 수정을 성공했습니다.

yummygyudon committed:
FEAT : Update 전용 Request Data 매핑 DTO 정의

- id 있는 상태의 Request에 맞춘 DTO
- JSON 속성명 Mapping 오류 @JsonProperty 어노테이션으로 해결

yummygyudon committed:
FEAT : ID 조회 메서드 정의 및 Update 메서드 정의

- 수정 Transaction 에 대한 서버측 에러 발생
  - `@Modifying` 을 통해 수정(update) 쿼리임을 명시하여 해결

yummygyudon committed:
FEAT : Retrospect 수정 Service 메서드 정의

- Request Body 데이터 Converting
- 수정 결과를 Response Body 에 담길 RetrospectResponseDto로 Converting

yummygyudon committed:
Retrospect 수정 PUT Request 매핑 Controller 정의

- Status : 200 OK
- 수정 결과 반환

yummygyudon commented:
#9 해당 이슈 기능 구현 Branch (`feat/#9-updateRetrospect`) Merge

funnysunny08 committed:
Merge branch 'main' into feat/#7-createRetospect
